docs/design.md

```diff
@@ -1,92 +1,122 @@
-#Design
+# Design
 
-##Architecture
+- [Architecture](#architecture)
+- [UI Component](#ui-component)
+- [Logic Component](#logic-component)
+- [Storage Component](#storage-component)
+- [Common Component](#common-component)
+- [Test Driver Component](#test-driver-component)
+- [Client Component](#client-component)
+
+## Architecture
 
 ![High Level Architecture](images/highlevelArchitecture.png)
 
-TEAMMATES is a Web app that runs on Google App Engine (GAE) cloud platform. Given above is an overview of the main components. 
-+ **UI**: The UI seen by users consists of Web pages containing HTML, CSS (for styling) and JavaScripts (for client-side interactions such as sorting, input validation etc.). This UI is generated by JavaServer Pages (JSP), using the JavaServer Pages Standard Tag Library (JSTL), and Java Servlets residing on the server. Requests are sent to the server over normal HTTP. In future, AJAX may be used sparingly to fetch data from the server asynchronously.
-+ **Logic**: The main logic of the app is in Java POJOs (Plain Old Java Objects). Some automated tasks are implemented as Servlets.
-+ **Storage**: Storage uses the persistence framework provided by GAE 'datastore', a noSQL database.
-+ **Test Driver**: TEAMMATES makes heavy use of automated regression testing. TestNG is used for Java unit testing and QUnit for JavaScript unit testing. We use Selenium Web Driver to automate UI testing. Test Driver can access the application directly to set up test data. We use JSON format to transmit such data to the application.
-+ **Client**: The Client component can connect to the back end directly without using a Browser. It is used for administrative purposes. e.g., migrating data to a new schema
-+ **Common**: The Common component contains utility code used across the application.
+TEAMMATES is a Web application that runs on Google App Engine (GAE) cloud platform. Given above is an overview of the main components.
```

Comment:
Should be either "**the** GAE cloud platform" or just "GAE" without "cloud platform"